### ISSUE TITLE
getting ceph demo running

### ISSUE THREAD

**Issue by ink1** [ceph/ceph-docker#79]
I'm interested in testing how ceph can be accessed through S3 API or, failing that, some other API. Any help would be much appreciated. It seems that the ceph demo starts RGW so using it could be an easy start. What are the steps to get the ceph-demo running? Specifically, am I right assuming that ceph/config has to be run (or config created) before running ceph/demo?

Unfortunately for me ceph/config hangs on "Configuration is locked by another host. Waiting." and ceph/demo immediately exits (no config?) although I have been able to run both with limited success in the past. This is what I'm trying to run now (etcd is running on the host itself)
```bash
docker run -e ETCDCTL_PEERS=http://172.16.X.Y:4001 -e MON_IP=172.16.X.Y \
-e MON_NAME=mymon -e CLUSTER=testing -v /etc/ceph:/etc/ceph ceph/config

docker run -d --net=host -v /etc/ceph:/etc/ceph -e MON_IP=172.16.X.Y \
-e CEPH_NETWORK=172.16.0.0/16 ceph/demo
```

**Comment by leseb** [comment]
Hey,

You don't need to run ceph/config, you can simply run the container like explained in the doc:

`docker run -d --net=host -v /etc/ceph:/etc/ceph -e MON_IP=192.168.0.20 -e CEPH_NETWORK=192.168.0.0/24 ceph/demo`

Here using `--net=host` is the most important thing.

**Comment by ink1** [comment]
Hi, 
I don't want to re-open the issue but just wanted to clarify what the issue was in case it can be useful for some one else. To begin with, I was running docker on Centos7 and subsequently on Ubuntu 14.04 (in a VM). There was no problem running ceph-demo on Ubuntu. Everything just worked, so thank you guys for all your good work. I noticed however that the demo populates /etc/ceph directory whereas it remained empty on Centos7. Knowing that Docker is not allowed to write to /etc/ceph by default I had changed SELinux security context
```bash
chcon -Rt svirt_sandbox_file_t /etc/ceph
```
but then subsequently deleted the folder and forgot to restore the context hence my problem.
Once the context was restored I was able to run ceph-demo exactly as I tried earlier 
```bash
docker run -d --net=host -v /etc/ceph:/etc/ceph -e MON_IP=172.16.X.Y \
-e CEPH_NETWORK=172.16.0.0/16 ceph/demo
```
and access it from the Centos7 host. Thanks!
